src/parser/ast.ml

```diff
@@ -13,21 +23,33 @@ type inline_element =
   | `Reference of
     reference_kind * string with_location * inline_element with_location list
   | `Link of string * inline_element with_location list ]
+(** Inline elements are equivalent to what would be found in a [span] in HTML.
+    Mostly these are straightforward. The [`Reference] constructor takes a triple
+    whose second element is the reference itself, and the third the replacement
+    text. Similarly the [`Link] constructor has the link itself as first parameter
+    and the second is the replacement text. *)
 
 type nestable_block_element =
   [ `Paragraph of inline_element with_location list
-  | `Code_block of string
+  | `Code_block of string option * string
```

Comment:
Some comments from https://github.com/ocaml/odoc/pull/678#discussion_r636941740

- Should we parse the "language" field ? (the first word)
  We might want to use it for syntax highlighting. This feature is targetted at Mdx, which uses this field.
- Both arguments should have a location attached.

Reply:
Good question. I've currently done the minimum possible here in the parser, in the spirit of how references are parsed, though this and the stripping of whitespace both seem reasonable to me.

Reply:
and I agree with the location comment, fill fix, thanks!